javatests/arcs/android/host/ArcHostHelperTest.kt

```diff
@@ -0,0 +1,186 @@
+    }
+
+    @Test
+    fun onStartCommand_doesNotCallOnStartArc_when_intentActionDoesNotMatch() = runBlockingTest {
+        helper.onStartCommand(Intent().apply { action = "Incorrect" })
+        assertThat(arcHost.startCalls()).isEmpty()
+    }
+
+    @Test
+    fun onStartCommand_doesNotCallOnStartArc_when_Operation_isMissing() = runBlockingTest {
+        helper.onStartCommand(Intent().apply { action = ArcHostHelper.ACTION_HOST_INTENT })
+        assertThat(arcHost.startCalls()).isEmpty()
+    }
+
+    @Test
+    fun onStartCommand_callsOnStartArcStopArc_whenStarsAlign() = runBlockingTest {
+        val personSchema = Schema(
+            listOf(SchemaName("Person")), SchemaFields(mapOf("name" to Text), emptyMap()),
+            SchemaDescription(),
+        "42"
+        )
+
+        val connectionSpec = HandleConnectionSpec(
+            VolatileStorageKey(ArcId.newForTest("foo"), "bar"),
+            EntityType(personSchema)
+        )
+
+        val particleSpec = ParticleSpec(
+            "FooParticle", "foo.bar.FooParticle", mapOf("foo" to connectionSpec)
+        )
+
+        val planPartition = PlanPartition("id", "FooHost", listOf(particleSpec))
+        val startIntent = planPartition.createStartArcHostIntent(
+            TestAndroidArcHostService::class.toComponentName(context)
+        )
+        val stopIntent = planPartition.createStopArcHostIntent(
+            TestAndroidArcHostService::class.toComponentName(context)
+        )
+        helper.onStartCommandSuspendable(startIntent)
+        assertThat(arcHost.startCalls()).containsExactly(planPartition)
+
+        helper.onStartCommandSuspendable(stopIntent)
+        assertThat(arcHost.stopCalls()).containsExactly(planPartition)
+    }
+
+    @Test
+    fun onStartCommand_callsGetParticles() = runBlockingTest {
+        val particleIdentifier = ParticleIdentifier("foo.bar.Baz")
+        val particleIdentifier2 = ParticleIdentifier("foo.bar.Baz2")
+
+        arcHost.registerParticle(particleIdentifier)
+        arcHost.registerParticle(particleIdentifier2)
+
+        val getParticlesIntent = TestAndroidArcHostService::class.toComponentName(context)
+            .createGetRegisteredParticlesIntent()
+
+        // Wait for async result
+        runBlocking {
+            suspendCoroutine<List<ParticleIdentifier>?> { coroutine ->
+                ArcHostHelper.setResultReceiver(
+                    getParticlesIntent, object : ResultReceiver(Handler()) {
+                        override fun onReceiveResult(resultCode: Int, resultData: Bundle?) {
+                            val particles =
+                                ArcHostHelper.getParticleIdentifierListResult(resultData)
+
+                            assertThat(particles).containsExactly(
+                                particleIdentifier, particleIdentifier2
+                            )
+                            coroutine.resume(particles)
+                        }
+                    })
```

Comment:
Newline before the closing paren, and place the `object : ResultReceiver(...) {` on a new line as well.